usr/lib/pkcs11/cca_stdll/cca_specific.c

```diff
@@ -272,99 +272,99 @@ static CK_RV cca_resolve_lib_sym(void *hdl)
-    dll_CSNDPKI = (CSNDPKI_t) dlsym(hdl, "CSNDPKI");
-    dll_CSNDDSG = (CSNDDSG_t) dlsym(hdl, "CSNDDSG");
-    dll_CSNDDSV = (CSNDDSV_t) dlsym(hdl, "CSNDDSV");
-    dll_CSNDKTC = (CSNDKTC_t) dlsym(hdl, "CSNDKTC");
-    dll_CSNDPKX = (CSNDPKX_t) dlsym(hdl, "CSNDPKX");
-    dll_CSNDSYI = (CSNDSYI_t) dlsym(hdl, "CSNDSYI");
-    dll_CSNDSYX = (CSNDSYX_t) dlsym(hdl, "CSNDSYX");
-    dll_CSUACFQ = (CSUACFQ_t) dlsym(hdl, "CSUACFQ");
-    dll_CSUACFC = (CSUACFC_t) dlsym(hdl, "CSUACFC");
-    dll_CSNDSBC = (CSNDSBC_t) dlsym(hdl, "CSNDSBC");
-    dll_CSNDSBD = (CSNDSBD_t) dlsym(hdl, "CSNDSBD");
-    dll_CSUALCT = (CSUALCT_t) dlsym(hdl, "CSUALCT");
-    dll_CSUAACM = (CSUAACM_t) dlsym(hdl, "CSUAACM");
-    dll_CSUAACI = (CSUAACI_t) dlsym(hdl, "CSUAACI");
-    dll_CSNDPKH = (CSNDPKH_t) dlsym(hdl, "CSNDPKH");
-    dll_CSNDPKR = (CSNDPKR_t) dlsym(hdl, "CSNDPKR");
-    dll_CSUAMKD = (CSUAMKD_t) dlsym(hdl, "CSUAMKD");
-    dll_CSNDRKD = (CSNDRKD_t) dlsym(hdl, "CSNDRKD");
-    dll_CSNDRKL = (CSNDRKL_t) dlsym(hdl, "CSNDRKL");
-    dll_CSNDSYG = (CSNDSYG_t) dlsym(hdl, "CSNDSYG");
-    dll_CSNBPTR = (CSNBPTR_t) dlsym(hdl, "CSNBPTR");
-    dll_CSNBCPE = (CSNBCPE_t) dlsym(hdl, "CSNBCPE");
-    dll_CSNBCPA = (CSNBCPA_t) dlsym(hdl, "CSNBCPA");
-    dll_CSNBPGN = (CSNBPGN_t) dlsym(hdl, "CSNBPGN");
-    dll_CSNBPVR = (CSNBPVR_t) dlsym(hdl, "CSNBPVR");
-    dll_CSNBDKG = (CSNBDKG_t) dlsym(hdl, "CSNBDKG");
-    dll_CSNBEPG = (CSNBEPG_t) dlsym(hdl, "CSNBEPG");
-    dll_CSNBCVE = (CSNBCVE_t) dlsym(hdl, "CSNBCVE");
-    dll_CSNBCSG = (CSNBCSG_t) dlsym(hdl, "CSNBCSG");
-    dll_CSNBCSV = (CSNBCSV_t) dlsym(hdl, "CSNBCSV");
-    dll_CSNBCVG = (CSNBCVG_t) dlsym(hdl, "CSNBCVG");
-    dll_CSNBKTP = (CSNBKTP_t) dlsym(hdl, "CSNBKTP");
-    dll_CSNDPKE = (CSNDPKE_t) dlsym(hdl, "CSNDPKE");
-    dll_CSNDPKD = (CSNDPKD_t) dlsym(hdl, "CSNDPKD");
-    dll_CSNBPEX = (CSNBPEX_t) dlsym(hdl, "CSNBPEX");
-    dll_CSNBPEXX = (CSNBPEXX_t) dlsym(hdl, "CSNBPEXX");
-    dll_CSUARNT = (CSUARNT_t) dlsym(hdl, "CSUARNT");
-    dll_CSNBCVT = (CSNBCVT_t) dlsym(hdl, "CSNBCVT");
-    dll_CSNBMDG = (CSNBMDG_t) dlsym(hdl, "CSNBMDG");
-    dll_CSUACRA = (CSUACRA_t) dlsym(hdl, "CSUACRA");
-    dll_CSUACRD = (CSUACRD_t) dlsym(hdl, "CSUACRD");
-    dll_CSNBTRV = (CSNBTRV_t) dlsym(hdl, "CSNBTRV");
-    dll_CSNBSKY = (CSNBSKY_t) dlsym(hdl, "CSNBSKY");
-    dll_CSNBSPN = (CSNBSPN_t) dlsym(hdl, "CSNBSPN");
-    dll_CSNBPCU = (CSNBPCU_t) dlsym(hdl, "CSNBPCU");
-    dll_CSUAPCV = (CSUAPCV_t) dlsym(hdl, "CSUAPCV");
-    dll_CSUAPRB = (CSUAPRB_t) dlsym(hdl, "CSUAPRB");
-    dll_CSUADHK = (CSUADHK_t) dlsym(hdl, "CSUADHK");
-    dll_CSUADHQ = (CSUADHQ_t) dlsym(hdl, "CSUADHQ");
-    dll_CSNDTBC = (CSNDTBC_t) dlsym(hdl, "CSNDTBC");
-    dll_CSNDRKX = (CSNDRKX_t) dlsym(hdl, "CSNDRKX");
-    dll_CSNBKET = (CSNBKET_t) dlsym(hdl, "CSNBKET");
-    dll_CSNBHMG = (CSNBHMG_t) dlsym(hdl, "CSNBHMG");
-    dll_CSNBHMV = (CSNBHMV_t) dlsym(hdl, "CSNBHMV");
+    *(void **)(&dll_CSNBCKI) = dlsym(hdl, "CSNBCKI");
```

Comment:
What is the benefit of casting all these to void* instead of using the type casts ?

Reply:
Its a workaround to assign a function pointer to a pointer in a warnig-free way. See man 3 dlopen. I added a reference to that man page in the commit message.